### ISSUE TITLE
create-react-app: PDF text is reversed when rendered / copying

### ISSUE THREAD

**Issue by asaldivar** [MatthewHerbst/react-to-print#347]
First of all, thank you for this GREAT library which is saving me so much time!

When I copy the example from the [demo](https://codesandbox.io/s/rzdhd?file=/example/index.js) within my app and my app is running `"react-to-print": "^2.12.3"` the pdf is generated seemingly perfectly. However, when I attempt to search for text within the PDF I'm unable to. After some digging I realized I'm unable to search for text because it is somehow reveresed/backwards/mirrored.

Note how I have to type the word _column_ backwards to be able to find it:
<img width="1149" alt="Screen Shot 2021-02-13 at 11 33 49 AM" src="https://user-images.githubusercontent.com/6180653/107859592-5a07bd00-6def-11eb-9e2e-4ab5201063ca.png">

If I type _column_ normally I get no results:
<img width="1150" alt="Screen Shot 2021-02-13 at 11 34 07 AM" src="https://user-images.githubusercontent.com/6180653/107859595-65f37f00-6def-11eb-8c2e-418ae8a787d9.png">

Further, when I copy and paste some of the text from the PDF it shows the text as reversed.

Also, I'm unable to highlight single characters from the PDF successfully:
<img width="439" alt="Screen Shot 2021-02-13 at 11 35 19 AM" src="https://user-images.githubusercontent.com/6180653/107859640-90453c80-6def-11eb-82d5-9061b1fcf265.png">

For extra clarity, this is the exact code I copied and am running within my app:
```js
import React from 'react';
import ReactToPrint from 'react-to-print';

class ComponentToPrint extends React.Component {
  render() {
    return (
      <table>
        <thead>
          <th>column 1</th>
          <th>column 2</th>
          <th>column 3</th>
        </thead>
        <tbody>
          <tr>
            <td>data 1</td>
            <td>data 2</td>
            <td>data 3</td>
          </tr>
          <tr>
            <td>data 1</td>
            <td>data 2</td>
            <td>data 3</td>
          </tr>
          <tr>
            <td>data 1</td>
            <td>data 2</td>
            <td>data 3</td>
          </tr>
        </tbody>
      </table>
    );
  }
}

class Example extends React.Component {
  render() {
    return (
      <div>
        <ReactToPrint
          trigger={() => <a href="#">Print this out!</a>}
          content={() => this.componentRef}
        />
        <ComponentToPrint ref={el => (this.componentRef = el)} />
      </div>
    );
  }
}

export default Example;
```

Thanks in advance for any guidance!

**Comment by MatthewHerbst** [comment]
Hello. I just printed from the demo and saved as PDF, and when I open the PDF I am able to search for text without issue. I used Chrome 88 to print on MacOS 11.12.1, using MacOS's default Preview app to view the PDF. What browser/OS are you on, and what tool are you using to view the PDF? (I think you are on MacOS and using Viewer as well from your pics, right?)

![Screen Shot 2021-02-13 at 11 48 07 AM](https://user-images.githubusercontent.com/1287375/107859903-6c82f600-6df1-11eb-974a-64e5176c74a6.png)

**Comment by asaldivar** [comment]
Thanks for the fast reply @MatthewHerbst ! Yes, when I print from the demo example (not within my own app) the searching for text also works perfectly. It's only from within my own app that the text becomes reversed in the PDF.

I am on Chrome version 88, macOS Big Sur 11.2, and using MacOS default Preview app to view the PDF.

My app is a simple Create React App. Is it possible there is some kind of setting that could be affecting the package?

**Comment by MatthewHerbst** [comment]
This is very strange. I am able to replicate the issue when using Create React App: https://codesandbox.io/s/create-react-app-forked-h4omm?file=/src/App.js

I'll do some digging here. It's clearly an issue with something `create-react-app` is doing, but not sure what. I'll get back to you, thanks for reporting

**Comment by asaldivar** [comment]
Thank you **so much** @MatthewHerbst . Really appreciate your efforts and looking forward to hearing from you. I'll keep digging as well and if I find anything out I'll report back ASAP!

**Comment by asaldivar** [comment]
@MatthewHerbst For further context, I found this [CRA app](https://github.com/lyhd/reactjs/tree/react-pdf) where the searching for text does work as expected. Note it's running [version](https://github.com/lyhd/reactjs/blob/react-pdf/package.json#L15) `"react-scripts": "3.4.1"`. I just tried fixing my app's CRA version from `^3.4.0` to `3.4.1` like in that repo but still faced the same issue. The CRA version you are running in your last linked codesandbox example is `"react-scripts": "2.0.5"`. This makes me wonder if it is a CRA caused issue or possibly something else?

**Comment by asaldivar** [comment]
@MatthewHerbst Please note that for the working example I linked, [CRA app](https://github.com/lyhd/reactjs/tree/react-pdf), you must be using the code from the `react-pdf` branch.

**Comment by asaldivar** [comment]
@MatthewHerbst, aha! I haven't quite figured out the _why_ but I have figured out the _what_. For some reason `react-to-print`'s functionality doesn't seem to like global styles. I forked your example and simply commenting out the `index.css` import fixes the text selectability within the PDF. [Here is that fork](https://codesandbox.io/s/create-react-app-forked-b70xp?file=/src/index.js:61-85).

I'm wondering if a change to how the styles are copied, maybe somewhere [here](https://github.com/gregnb/react-to-print/blob/master/src/index.tsx#L225) or [possibly here](https://github.com/gregnb/react-to-print/blob/master/src/index.tsx#L364), would fix the problem?

**Comment by MatthewHerbst** [comment]
@asaldivar thanks so much for investigating! I'm not sure exactly how global styles could be causing an issue. I'll keep playing around with it. I'll have more time tomorrow to look at it. Please let me know if you discover anything else, thank you!

**Comment by MatthewHerbst** [comment]
Hi @asaldivar my friend and I were looking at this and we believe it comes down to a mix of Chrome and the `BlinkMacSystemFont` font. When the page is printed via Firefox there is no issue. On Chrome, if you remove the `BlinkMacSystemFont` font there is no issue.

It _might_ be related to [this Chromium bug](https://bugs.chromium.org/p/chromium/issues/detail?id=1018581), but not sure. I'll keep researching that, but for now just avoiding use of the `BlinkMacSystemFont` should fix your problems

**Comment by asaldivar** [comment]
@MatthewHerbst Thanks so much!!!! Will try out your suggestion and if it doesn't go as expected will report back to you. Thanks again for your effort and kudos on such a great package!

**Comment by mcdenhoed** [comment]
A bit more info on this issue:

I just ran into the same issue, but the PDF and original page were NOT using `BlinkMacSystemFont`. Additionally it manifested when looking at the PDF in macOS Preview, but not in Adobe Acrobat.

After some testing, I think it was because I was using a Variable Font. I replaced the Variable Font with the static-weight versions of the weights I was using and that fixed it!
